### ISSUE TITLE
env var

### ISSUE THREAD

**Issue by nrc04** [peterzen/hass-pws-mqtt-addon#29]
Hello,

Have new problem, do not understand :
2024/01/08 19:56:30 PWS_IP env var undefined

I had uninstall it and try many time no changes.

thanks
regards

**Comment by peterzen** [comment]
Hi, can you please confirm which addon version is running on your HA?

**Comment by wilburne** [comment]
version de l'addon : 0.1.23

**Comment by peterzen** [comment]
Right, I see the problem now.  Can you please downgrade to 0.1.22?  

- uninstall the addon
- remove the repository
![image](https://github.com/peterzen/hass-pws-mqtt-addon/assets/5878500/a7c29ce0-04c8-40b5-a6a6-455007e0558d)

- add it back again: https://github.com/peterzen/hass-pws-mqtt-addon
- reinstall the addon again.

Apologies for the problem!

**Comment by wilburne** [comment]
Bonsoir,
merci pour la réponse, j'ai effectué la nouvelle installation, mais je n'arrive toujours pas à collecter les données.
J'ai ce message lorsque je démarre pws:

"PWS to MQTT dispatcher addon
_s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun pwsmqttdispatcher (no readiness notification)
s6-rc: info: service legacy-services successfully started
[18:54:14] INFO: Starting pwsmqttdispatcher
./run: line 21: /usr/bin/pwsmqttdispatcher: cannot execute binary file: Exec format error
./run: line 21: /usr/bin/pwsmqttdispatcher: No error information
[17:54:14] WARNING: Halt add-on with exit code 1
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped"_

Une idée?
Remerciements
Cdlt

**Comment by peterzen** [comment]
Are you running on a raspberry Pi, or a PC?
A fix was released last night, please make sure you have 0.1.24 installed.

**Comment by peterzen** [comment]
Apparently the yesterday release broke the Raspberry build.  Until it gets fixed you should downgrade back to the previous version with the steps above.  I apologize for this!

**Comment by wilburne** [comment]
Good evening,
you tell me to reinstall the previous version.
How can I put version v0.1.16 or v0.1.21
thanks for the help
Sincerely

**Comment by peterzen** [comment]
I fixed and tested 0.1.24 on raspberry pi 3 earlier today so you can update to 0.1.25.  Click "Check updates" in the Addons screen, it should highlight the PWS addon indicating that there is an update available.  Try restarting HA if that's not the case.  If the update still doesn't show up (this sometimes happens to me) then remove the addon and the repo (as per the above steps with the screenshots) and reinstall it.

**Comment by wilburne** [comment]
ok update done, i am now in 0.1.25

![2024-01-10 190755 pws](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/3c96cce7-af86-4c1c-a795-2ebbb47be4ad)
![2024-01-10 190755 mqtt](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/befcdeb1-ccc3-4ed6-9e92-40a333c8320f)
![2024-01-10 190755 config](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/d155ddb8-3874-40f0-9e76-53deaf6bb92b)
![2024-01-10 190755](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/3b6e91d6-aa2b-4144-a0b2-1a45b085443e)

publishing ok, sensor ok, but not found by mqtt brocker, no incoming messages

**Comment by peterzen** [comment]
It looks like the publishing to the `weather_station` topic works.  Can you check with the MQTT explorer addon whether the data appears there? https://community.home-assistant.io/t/addon-mqtt-explorer-new-version/603739

**Comment by peterzen** [comment]
What is the addon from the last screenshot?

**Comment by peterzen** [comment]
You should be seeing this:
![image](https://github.com/peterzen/hass-pws-mqtt-addon/assets/5878500/be2b3b72-2047-40c3-8385-d578e276e1aa)

**Comment by wilburne** [comment]
addon : 0.1.25
see  log mosquitto broker, denied publish.
![2024-01-10 193846 ver addon](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/013decd3-83eb-4c2e-bb9d-4dab6b443d5a)
![2024-01-10 193636 Mosquitto broker](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/ca2bf3e6-c398-4936-b54b-23fdb2e06d83)

**Comment by wilburne** [comment]
2024-01-10 19:55:57: Denied PUBLISH from pwsmqttdispatcher (d0, q0, r0, m0, 'weather_station', ... (441 bytes))

not able to use mqttexplorer

**Comment by wilburne** [comment]
![2024-01-10 201140 mqtt explorer](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/d0be4a21-b707-4a54-acd9-3d92047e01cf)

no publication

**Comment by wilburne** [comment]
for your inforemation, I create a new user with 
id:pwsmqttdispatcher 
and 
password : pwsmqttdispatcher

**Comment by peterzen** [comment]
Did this change since 0.1.21?  MQTT auth is managed by mosquitto, the PWS addon picks up the default credentials from HA.  Did you try a full HA restart?

**Comment by wilburne** [comment]
yes I restart but I am still denied in log file
![2024-01-11 100556 mosquito](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/f3673d51-85b1-40b7-a2b2-03aa77b638e6)
regards

**Comment by peterzen** [comment]
This sounds like a configuration issue in the broker as the PWS addon successfully pushes the data to MQTT.  That part of the code has not changed in the recent version, it works the same way as in 0.1.21.

**Comment by wilburne** [comment]
Good evening,
I agree with you, but look at my brooker setup:
I don't see what could be wrong.
Greetings
![Mosquitto broker config 2](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/68b409ac-bcfc-4926-8f32-df00b9369671)
![Mosquitto broker config 1](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/4137a418-866a-455f-a232-3ad15284d884)

**Comment by peterzen** [comment]
The only thing that's different from my setup is the `mqtt_topic` setting, can you try this PWS config instead?

```yaml
pws_ip: x.x.x.x
mqtt_topic: weather_station/data
mqtt_client_id: pwsmqttdispatcher
fetch_interval: 180
debug_enabled: true
```

**Comment by wilburne** [comment]
setting like you
no change
![2024-01-12 195039 mosquitto](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/e592ef28-0fde-4a34-9508-3e101f029e4d)
![2024-01-12 194720 pws new config](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/c9e66021-6cac-49af-86f9-f78711ac5810)

**Comment by peterzen** [comment]
Can you try setting `mqtt_topic` to `PVROUTER005/DATA` and see what happens?

**Comment by wilburne** [comment]
mqtt_topic PVROUTER005/DATA publish without problem

**Comment by peterzen** [comment]
What addon is this screenshot from?  I wonder if it might have something to do with it.  What is sending to PVROUTER005/DATA?

![image](https://github.com/peterzen/hass-pws-mqtt-addon/assets/5878500/3785f5bb-cc8a-4cd5-ac5a-0e0470b9f24b)

**Comment by wilburne** [comment]
n MQTT I have 2 devices.
pws which is my MTO WH2600 station
And
PVROUTER005 which is a module that manages the water heaters in my home
![2024-01-12 213747](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/b5fb4e01-fc64-4249-9e9f-efb49b24d061)

**Comment by peterzen** [comment]
What is the screenshot in the previous post from?

**Comment by wilburne** [comment]
in "Devices and service"\MQTT\Devices, I select "wh2600" then "MQTT info"

**Comment by peterzen** [comment]
> in "Devices and service"\MQTT\Devices, I select "wh2600" then "MQTT info"

Interesting, I don't have my device configured this way.  Maybe you could doublecheck the subscribed topic variable, it looks like it's set to `weather_station` above, not `weather_station/data`

**Comment by wilburne** [comment]
exact, but I had previously tried "weather-station" but still the same problem, and following your post I modified it again to "weater_station/data".
It does not work either.

**Comment by peterzen** [comment]
It can be any value really.  I'd try changing it to something else, surely the problem is around the topic configuration, quite possibly in mosquitto.

**Comment by wilburne** [comment]
Good morning
yesss,
I finally managed to read the data.
Indeed, I just changed the value of the Mosquitto broker configuration, from "customize" active: true to "customize active: false. (see attached files)
Once done, the transfer of information takes place without problem.
Don't ask me why, I don't know anything about it.
Thank you for your help and have a good evening.
Regards
![Mosquitto broker config false](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/d6f57c29-cdff-4194-978e-9491f83c01de)
![Mosquitto broker config true](https://github.com/peterzen/hass-pws-mqtt-addon/assets/156028340/922e268c-c7c6-4b7b-820c-f635fa104d4f)
